### PR TITLE
fix(FloatButton): fix FloatButton menu mode to allow interaction inside the menu on hover

### LINKED ISSUE #51158: FloatButton 悬浮按钮的菜单模式，触发条件是hover的情况下，无法点击菜单里面的图标
### Reproduction link

[https://ant-design.antgroup.com/components/float-button-cn#float-button-demo-group-menu](https://ant-design.antgroup.com/components/float-button-cn#float-button-demo-group-menu)

### Steps to reproduce

直接使用官方的例子试一下就可以，很难点击到菜单里面的2个图标，只要鼠标移动出悬浮按钮的范围，上面的菜单就收起来了，之前使用5.18.1版本的时候没有这个问题

### What is expected?

期望点击菜单按钮或者点击别的地方才关闭菜单

### What is actually happening?

菜单自动关闭，无法点击菜单内的图片

| Environment | Info |
| --- | --- |
| antd | 5.21.2 |
| React | 18.3.1 |
| System | macOS 15.0.1 |
| Browser | chrome129 |

### PR DESCRIPTION
**Pull request by aojunhao123**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues
fix #51158 

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution
**Background:**
In the FloatButton menu mode, when the trigger condition is set to hover, the menu automatically closes as soon as the mouse leaves the button, making it difficult to interact with the icons inside the menu. This issue was not present in version 5.18.1.

**Solution:**
We implemented an invisible buffer area between the FloatButton and its pop-up menu. This buffer creates a "safe zone" where users can move their mouse from the button to the menu without triggering a close event. The buffer's size and position are dynamically set based on the menu's placement, ensuring a smooth transition for users to interact with the menu items without the menu disappearing prematurely.

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fixed the issue where the FloatButton menu mode closed prematurely on hover, preventing interaction with the menu icons. |
| 🇨🇳 Chinese | 修复了 FloatButton 菜单模式下，悬浮时菜单过早关闭，导致无法点击菜单图标的问题。 |

### CODE REVIEW

File: components/float-button/FloatButtonGroup.tsx [hunk 1]
```diff
@@ -139,7 +158,11 @@ const FloatButtonGroup: React.FC<Readonly<FloatButtonGroupProps>> = (props) => {
           <>
             <CSSMotion visible={open} motionName={`${groupPrefixCls}-wrap`}>
               {({ className: motionClassName }) => (
-                <div className={classNames(motionClassName, wrapperCls)}>{children}</div>
+                <div className={classNames(motionClassName, wrapperCls)}>
+                  {children}
+                  {/* buffer area */}
+                  <div style={bufferStyle} />
```
Comment (afc163) [inline components/float-button/FloatButtonGroup.tsx:164]: 1. 用伪元素实现。
2. 用 style 样式代码实现，不放在 js 里。

Comment (aojunhao123) [reply components/float-button/FloatButtonGroup.tsx:164]: 好的

File: components/float-button/FloatButtonGroup.tsx [hunk 2]
```diff
@@ -139,7 +139,11 @@ const FloatButtonGroup: React.FC<Readonly<FloatButtonGroupProps>> = (props) => {
           <>
             <CSSMotion visible={open} motionName={`${groupPrefixCls}-wrap`}>
               {({ className: motionClassName }) => (
-                <div className={classNames(motionClassName, wrapperCls)}>{children}</div>
+                <div
+                  className={classNames(motionClassName, wrapperCls, `${groupPrefixCls}-buffer`)}
```
Comment (afc163) [inline components/float-button/FloatButtonGroup.tsx:143]: buffer 这个类目是多余的？直接放 wrapperCls 上。

File: components/float-button/style/index.ts [hunk 3]
```diff
@@ -246,6 +246,37 @@ const floatButtonGroupStyle: GenerateStyle<FloatButtonToken, CSSObject> = (token
...
+        content: '""',
+        position: 'absolute',
+        zIndex: 1,
+      },
+    },
+    [`${groupPrefixCls}-top > ${groupPrefixCls}-buffer::after`]: {
+      left: '-20px',
+      right: '-20px',
+      height: '20px',
+      bottom: '-20px',
```
Comment (afc163) [inline components/float-button/style/index.ts:260]: 20 应该复用已有的样式变量。

File: components/float-button/style/index.ts [hunk 4]
```diff
@@ -246,6 +246,37 @@ const floatButtonGroupStyle: GenerateStyle<FloatButtonToken, CSSObject> = (token
...
     },
+    [`${groupPrefixCls}-buffer`]: {
+      '&::after': {
+        content: '""',
+        position: 'absolute',
+        zIndex: 1,
+      },
+    },
+    [`${groupPrefixCls}-top > ${groupPrefixCls}-buffer::after`]: {
+      left: '-20px',
```
Comment (li-jia-nan) [inline components/float-button/style/index.ts:257]: 这四个属性用 inset 代替就好了吧？
image

File: components/float-button/style/index.ts (hunk 3 above)
Comment (li-jia-nan) [reply components/float-button/style/index.ts:260]: 这个 20 是怎么算出来的？

File: components/float-button/style/index.ts [hunk 5]
```diff
@@ -136,21 +136,53 @@ const floatButtonGroupStyle: GenerateStyle<FloatButtonToken, CSSObject> = (token
...
       flexDirection: 'row',
       left: { _skip_check_: true, value: calc(floatButtonSize).add(margin).equal() },
       right: { _skip_check_: true, value: 'auto' },
+      '&::after': {
+        content: '""',
+        position: 'absolute',
+        zIndex: 1,
+        width: margin,
+        height: '100%',
+        left: calc(margin).mul(-1).equal(),
```
Comment (li-jia-nan) [inline components/float-button/style/index.ts:171]: left 和 right 这样应该会报错，要么用逻辑属性，要么像上面那样跳过检查

Comment (aojunhao123) [reply components/float-button/style/index.ts:171]: 好的，那我再去修改一下；顺便问一下这个报错的原因是什么呢

Comment (li-jia-nan) [reply components/float-button/style/index.ts:171]: antd 中禁止直接使用方向属性，为了适配 LTR 和 RTL 规则，所以这里需要跳过校验，参考上面的写法

File: components/float-button/style/index.ts [hunk 6]
```diff
@@ -142,7 +142,7 @@ const floatButtonGroupStyle: GenerateStyle<FloatButtonToken, CSSObject> = (token
         zIndex: 1,
         width: '100%',
         height: margin,
-        bottom: calc(margin).mul(-1).equal(),
+        bottom: { _skip_check_: true, value: calc(margin).mul(-1).equal() },
```
Comment (li-jia-nan) [inline components/float-button/style/index.ts:145]: top bottom 不用 _skip_check_，只是左右跳过检查就好了

Comment (aojunhao123) [reply components/float-button/style/index.ts:145]: 嗯嗯了解了（

Comment (aojunhao123) [reply components/float-button/style/index.ts]: 顺便想问一下可否增加一个切换触发方式（hover/trigger）的下拉框，算是对文档demo的一个小小补充（
QQ_1728897081211

Comment (afc163) [reply components/float-button/style/index.ts]: 没必要，trigger 有单独的 demo。

Comment (aojunhao123) [reply components/float-button/style/index.ts]: 好的

File: components/float-button/style/index.ts [hunk 7]
```diff
@@ -136,21 +136,53 @@ const floatButtonGroupStyle: GenerateStyle<FloatButtonToken, CSSObject> = (token
...
     [`${groupPrefixCls}-bottom > ${groupPrefixCls}-wrap`]: {
       flexDirection: 'column',
       top: calc(floatButtonSize).add(margin).equal(),
       bottom: 'auto',
+      '&::after': {
+        content: '""',
+        position: 'absolute',
+        zIndex: 1,
+        width: '100%',
+        height: margin,
```
Comment (afc163) [inline components/float-button/style/index.ts:157]: 这一排，重复了四遍。

Comment (aojunhao123) [reply components/float-button/style/index.ts:157]: 前三个属性可以单独抽出来，但是缓冲区的width和height是根据placement是水平还是垂直来设置的

File: components/float-button/style/index.ts [hunk 8]
```diff
@@ -136,21 +136,53 @@ const floatButtonGroupStyle: GenerateStyle<FloatButtonToken, CSSObject> = (token
       flexDirection: 'column',
       top: 'auto',
       bottom: calc(floatButtonSize).add(margin).equal(),
+      '&::after': {
+        content: '""',
+        position: 'absolute',
+        zIndex: 1,
+        width: '100%',
+        height: margin,
```
[the remark this answers is not in the document]
Comment (aojunhao123) [reply components/float-button/style/index.ts:144]: 加上单位是指，我是参照其他地方的写法来的

[the remark this answers is not in the document]
Comment (li-jia-nan) [reply components/float-button/style/index.ts:147]: 哦，这里不用加，是我搞错了

File: components/float-button/style/index.ts [hunk 9]
```diff
@@ -101,6 +101,11 @@ const floatButtonGroupStyle: GenerateStyle<FloatButtonToken, CSSObject> = (token
     calc,
   } = token;
   const groupPrefixCls = `${componentCls}-group`;
+  const commonBufferStyle: CSSObject = {
```
Comment (li-jia-nan) [inline components/float-button/style/index.ts:104]: 这个对象是常量，可以提到最外面去，不用每次渲染都声明一遍

Comment (aojunhao123) [reply components/float-button/style/index.ts:104]: 疏忽了，这就去改（

File: components/float-button/style/index.ts [hunk 10]
```diff
@@ -86,6 +86,12 @@ export type FloatButtonToken = FullToken<'FloatButton'> & {
   floatButtonInsetInlineEnd: number;
 };
 
+const sharedBufferStyle: CSSObject = {
+  content: '""',
+  position: 'absolute',
+  zIndex: 1,
```
Comment (afc163) [inline components/float-button/style/index.ts:92]: 貌似这个 zIndex 不需要。

Comment (aojunhao123) [reply components/float-button/style/index.ts:92]: 好的，我再去看一下

File: components/float-button/style/index.ts [hunk 11]
```diff
@@ -86,6 +86,11 @@ export type FloatButtonToken = FullToken<'FloatButton'> & {
   floatButtonInsetInlineEnd: number;
 };
 
+const sharedBufferStyle: CSSObject = {
```
Comment (li-jia-nan) [inline components/float-button/style/index.ts:89]:
```suggestion
const sharedBufferStyle: React.CSSProperties = {
```

Comment (aojunhao123) [reply components/float-button/style/index.ts]: 已修改

File: components/float-button/style/index.ts [hunk 12]
```diff
@@ -86,6 +86,11 @@ export type FloatButtonToken = FullToken<'FloatButton'> & {
   floatButtonInsetInlineEnd: number;
 };
 
+const sharedBufferStyle: React.CSSProperties = {
+  content: '""',
+  position: 'absolute',
+};
```
Comment (afc163) [inline components/float-button/style/index.ts:92]: 只有两个 key 的话可复用的话，这个变量可能有点不划算了，要不去掉吧。

Comment (aojunhao123) [reply components/float-button/style/index.ts:92]: 好的😶

**Review by afc163 — APPROVED** [review]

**Pull request merged**